docs/understanding-airbyte/jobs.md

```diff
@@ -1,45 +1,111 @@
 
-## Worker Lifecycle
+### Worker-Job Architecture
 
-This section will depict the lifecycle of a worker. It will only show the 2 connector version. The single connector version is the same with one side removed.
+This section will depict the worker-job architecture as discussed above. Only the 2-connector version is shown. The single connector version is the same with one side removed.
 
-Note: When a source has passed all of its messages, the docker process should automatically exit. After a destination has received all records, it should automatically shutdown. The worker gives each a grace period to shutdown on their own. If that grace period expires, then the worker will force shutdown.
+The source process should automatically exit after passing all of its messages. Similarly, the destination process shutdowns after receiving all records. Each process is given a shutdown grace period. The worker forces shutdown if this is exceeded.
 
 ![Worker Lifecycle](../.gitbook/assets/worker-lifecycle.png)
 
 [Image Source](https://docs.google.com/drawings/d/1k4v_m2M5o2UUoNlYM7mwtZicRkQgoGLgb3eTOVH8QFo/edit)
 
 See the [architecture overview](high-level-view.md) for more information about workers.
 
-## Worker parallelization
+## Deployment Types
+
+Up to now, the term 'processes' has been used loosely. This section will describe this in more detail.
+
+Airbyte offers two deployment types. The underlying process implementations differ accordingly.
+
+1. The Docker deployment - Each process is a local process backed by a Docker container. As all processes are local, process communication is per standard unix pipes.
+2. The Kubernetes deployment - Each process is a backed by a Kubernetes pod. As Kubernetes does not make process-locality guarantees, Airbyte has implemented mechanisms to hide the remote process execution.
+   See [this blogpost](https://airbyte.com/blog/scaling-data-pipelines-kubernetes) for more details.
+
+### Decoupling Worker and Job Processes
+
+Workers being responsible for all non-connector-related job operations means multiple jobs are operationally dependent on a single worker process.
+
+There are two downsides to this:
+1. Any issues to the parent worker process affects all job processes launched by the worker.
+2. Unnecessary complexity of vertically scaling the worker process to deal with IO and processing requirements from multiple jobs.
+
+This gives us a potentially brittle system component that can be operationally tricky to manage. For example, since redeploying Airbyte terminates all worker processes, all running jobs are also terminated.
+
+The Container Orchestrator was introduced to solve this.
+
+#### Container Orchestrator
+When enabled, workers launch the Container Orchestrator process.
+
+This orchestrator process replaces the previous worker process. It now performs the [above listed responsibilities](#worker-responsibilities).
```

Comment:
I wonder if `replaces` can be misleading here. I think the worker still waits for the Orchestrator while I translate _replace_ as _a fire and forget_ type of operation.
What about `the worker process will delegate the [above listed responsibilities](#worker-responsibilities) to the orchestrator process`?

Reply:
I like delegate